Explorer/Assets/DCL/UI/MainUIContainer/HideShowView.cs

```diff
@@ -0,0 +1,131 @@
+        internal float animationTime { get; private set; }
+
+        [field: SerializeField]
+        internal PointerDetectionArea pointerDetectionArea { get; private set; }
+        private bool forceOpen;
+        private CancellationTokenSource hideCancellationTokenSource = new ();
+
+        private CancellationTokenSource showCancellationTokenSource = new ();
+        private bool showing;
+        private bool waitingToHide;
+
+        private bool waitingToShow;
+
+        private void Start()
+        {
+            showing = true;
+            pointerDetectionArea.OnEnterArea += OnPointerOnEnter;
+            pointerDetectionArea.OnExitArea += OnPointerOnExit;
+        }
+
+        public void SetForceOpen(bool value)
+        {
+            forceOpen = value;
+            hideCancellationTokenSource = hideCancellationTokenSource.SafeRestart();
+            showCancellationTokenSource = showCancellationTokenSource.SafeRestart();
+        }
+
+        private void OnPointerOnEnter()
+        {
+            if (forceOpen) return;
+
+            if (showing) { waitingToShow = false; }
+
+            if (showCancellationTokenSource.IsCancellationRequested) { waitingToShow = false; }
+
+            if (!showing && !waitingToShow)
+            {
+                waitingToShow = true;
+                showCancellationTokenSource = showCancellationTokenSource.SafeRestart();
+                WaitAndShow(showCancellationTokenSource.Token).Forget();
+            }
+
+            if (waitingToHide) { hideCancellationTokenSource.Cancel(); }
+        }
+
+        private void OnPointerOnExit()
+        {
+            if (forceOpen) return;
+
+            if (waitingToShow || showing) { showCancellationTokenSource.Cancel(); }
+
+            if (hideCancellationTokenSource.IsCancellationRequested) { waitingToHide = false; }
+
+            if (!waitingToHide && showing)
+            {
+                waitingToHide = true;
+                hideCancellationTokenSource = hideCancellationTokenSource.SafeRestart();
+                WaitAndHide(hideCancellationTokenSource.Token).Forget();
+            }
+        }
+
+        private async UniTask WaitAndHide(CancellationToken ct)
+        {
+            await UniTask.Delay(TimeSpan.FromSeconds(hideWaitTime), cancellationToken: ct);
+            waitingToHide = false;
+            if (ct.IsCancellationRequested) return;
+
+            await AnimateWidthAsync(hideLayoutWidth, ct);
+
+            if (ct.IsCancellationRequested) return;
+            showing = false;
+        }
+
+        private async UniTask WaitAndShow(CancellationToken ct)
+        {
+            await UniTask.Delay(TimeSpan.FromSeconds(showWaitTime), cancellationToken: ct);
+            waitingToShow = false;
+            if (ct.IsCancellationRequested) return;
+
+            await AnimateWidthAsync(showLayoutWidth, ct);
+
+            if (ct.IsCancellationRequested) return;
+            showing = true;
+        }
```

Comment:
The logic of these 2 functions are almost identical, we could extract it into a single function and avoid to repeat code.